### ISSUE TITLE
Cypher query fails with `Expected ... Instead, got: `

### ISSUE THREAD

**Issue by ontologiae** [neo4j/neo4j#9952]
Hi,

A Cypher query fails to execute and throw an error.
I got a long message with "Expected" followed by the scala version of the AST of the query, followed by "Instead, I got" with another long AST (see below).

**Neo4j Version:** 3.2.3
**Operating System:** Linux neo4j.dev 3.16.0-4-amd64 #1 SMP Debian 3.16.7-ckt25-1 (2016-03-06) x86_64 GNU/Linux
**API:** Cypher

### Steps to reproduce
1. Pull the image:[QueryFailedExpectedInsteadGot.zip](https://github.com/neo4j/neo4j/files/1267244/QueryFailedExpectedInsteadGot.zip)
2. Unzip the file and go to the `QueryFailedExpectedInsteadGot` directory
3. $ cypher-shell -a localhost -u neo4j -p YOURPASS --debug --format verbose <  CypherQueryBugDatas.cypher
4. cypher-shell -a localhost -u neo4j -p YOURPASS --debug --format verbose < QueryFailed.cypher

### Expected behavior
The query returns 0 or more lines

### Actual behavior

Got this : 
```
Expected 
RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED63),(IdName(setters),IdName(  UNNAMED74)),OUTGOING,List(RelTypeName(vBody)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED111),(IdName(  UNNAMED74),IdName(statements)),OUTGOING,List(RelTypeName(vstatements)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED151),(IdName(setters),IdName(  UNNAMED167)),OUTGOING,List(RelTypeName(vModifiers)),SimplePatternLength)),Set(IdName(setters), IdName(  UNNAMED74), IdName(statements), IdName(  UNNAMED167)),Set(),Selections(Set(Predicate(Set(IdName(  UNNAMED167)),HasLabels(Variable(  UNNAMED167),List(LabelName(Modifier)))), Predicate(Set(IdName(  UNNAMED74)),In(Property(Variable(  UNNAMED74),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING1,String))))), Predicate(Set(IdName(setters)),In(Property(Variable(setters),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING0,String))))), Predicate(Set(IdName(  UNNAMED167)),In(Property(Variable(  UNNAMED167),PropertyKeyName(isPublic)),ListLiteral(List(Parameter(  AUTOSTRING2,String))))), Predicate(Set(IdName(  UNNAMED74)),HasLabels(Variable(  UNNAMED74),List(LabelName(Block)))), Predicate(Set(IdName(setters)),HasLabels(Variable(setters),List(LabelName(MethodDeclaration)))))),Vector(),Set(),Set(),List()),AggregatingQueryProjection(Map(setters -> Variable(setters)),Map(statementsCount -> FunctionInvocation(Namespace(List()),FunctionName(count),false,Vector(Variable(statements)))),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(setters), IdName(statementsCount)),Selections(Set()),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(setters -> Variable(setters), statementsCount -> Variable(statementsCount),   FRESHID269 -> Equals(Variable(statementsCount),Parameter(  AUTOINT3,Integer))),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED289),(IdName(setters),IdName(  UNNAMED300)),OUTGOING,List(RelTypeName(vBody)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED337),(IdName(  UNNAMED300),IdName(  UNNAMED354)),OUTGOING,List(RelTypeName(vstatements)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED405),(IdName(  UNNAMED354),IdName(  UNNAMED408)),OUTGOING,List(),SimplePatternLength), PatternRelationship(IdName(  UNNAMED478),(IdName(setters),IdName(  UNNAMED495)),OUTGOING,List(RelTypeName(vparameters)),SimplePatternLength)),Set(IdName(  res@517), IdName(  UNNAMED495), IdName(  UNNAMED354), IdName(  UNNAMED300), IdName(  UNNAMED408), IdName(setters), IdName(  rankf@668)),Set(IdName(setters), IdName(statementsCount), IdName(  FRESHID269)),Selections(Set(Predicate(Set(IdName(  FRESHID269)),Variable(  FRESHID269)), Predicate(Set(IdName(setters), IdName(  res@517)),Equals(Property(Variable(setters),PropertyKeyName(methodName)),Property(Variable(  res@517),PropertyKeyName(methodName)))), Predicate(Set(IdName(  res@517), IdName(  rankf@668)),Equals(Property(Variable(  res@517),PropertyKeyName(uidFonctionEnglobante)),Property(Variable(  rankf@668),PropertyKeyName(uid)))), Predicate(Set(IdName(  UNNAMED289), IdName(  UNNAMED405)),Not(Equals(Variable(  UNNAMED289),Variable(  UNNAMED405)))), Predicate(Set(IdName(  UNNAMED354)),In(Property(Variable(  UNNAMED354),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING5,String))))), Predicate(Set(IdName(  rankf@668)),In(Property(Variable(  rankf@668),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING8,String))))), Predicate(Set(IdName(setters), IdName(  res@517)),Equals(Property(Variable(setters),PropertyKeyName(nomFichier)),Property(Variable(  res@517),PropertyKeyName(nomFichier)))), Predicate(Set(IdName(  UNNAMED300)),HasLabels(Variable(  UNNAMED300),List(LabelName(Block)))), Predicate(Set(IdName(  rankf@668)),HasLabels(Variable(  rankf@668),List(LabelName(MethodDeclaration)))), Predicate(Set(IdName(  res@517)),In(Property(Variable(  res@517),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING7,String))))), Predicate(Set(IdName(  UNNAMED408)),HasLabels(Variable(  UNNAMED408),List(LabelName(Assignment)))), Predicate(Set(IdName(  UNNAMED354)),HasLabels(Variable(  UNNAMED354),List(LabelName(ExpressionStatement)))), Predicate(Set(IdName(  res@517)),HasLabels(Variable(  res@517),List(LabelName(MethodInvocation)))), Predicate(Set(IdName(  UNNAMED408)),In(Property(Variable(  UNNAMED408),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING6,String))))), Predicate(Set(IdName(  UNNAMED300)),In(Property(Variable(  UNNAMED300),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING4,String))))), Predicate(Set(IdName(  UNNAMED337), IdName(  UNNAMED405)),Not(Equals(Variable(  UNNAMED337),Variable(  UNNAMED405)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(  res@517 -> Variable(  res@517),   rankf@668 -> Variable(  rankf@668),   ruleCoef@789 -> Parameter(  AUTOINT9,Integer)),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(  res@517), IdName(  rankf@668), IdName(  ruleCoef@789)),Selections(Set()),Vector(),Set(),Set(),List()),AggregatingQueryProjection(Map(res.nomFichier -> Property(Variable(  res@517),PropertyKeyName(nomFichier)), res.lineNum -> Property(Variable(  res@517),PropertyKeyName(lineNum)), 1 -> Parameter(  AUTOINT10,Integer)),Map(),QueryShuffle(List(),None,None)),None))))))) 


Instead, got: 
RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED151),(IdName(setters),IdName(  UNNAMED167)),OUTGOING,List(RelTypeName(vModifiers)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED63),(IdName(setters),IdName(  UNNAMED74)),OUTGOING,List(RelTypeName(vBody)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED111),(IdName(  UNNAMED74),IdName(statements)),OUTGOING,List(RelTypeName(vstatements)),SimplePatternLength)),Set(IdName(setters), IdName(  UNNAMED167), IdName(  UNNAMED74), IdName(statements)),Set(),Selections(Set(Predicate(Set(IdName(  UNNAMED167)),HasLabels(Variable(  UNNAMED167),List(LabelName(Modifier)))), Predicate(Set(IdName(  UNNAMED74)),In(Property(Variable(  UNNAMED74),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING1,String))))), Predicate(Set(IdName(setters)),In(Property(Variable(setters),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING0,String))))), Predicate(Set(IdName(  UNNAMED167)),In(Property(Variable(  UNNAMED167),PropertyKeyName(isPublic)),ListLiteral(List(Parameter(  AUTOSTRING2,String))))), Predicate(Set(IdName(  UNNAMED74)),HasLabels(Variable(  UNNAMED74),List(LabelName(Block)))), Predicate(Set(IdName(setters)),HasLabels(Variable(setters),List(LabelName(MethodDeclaration)))))),Vector(),Set(),Set(),List()),AggregatingQueryProjection(Map(setters -> Variable(setters)),Map(statementsCount -> FunctionInvocation(Namespace(List()),FunctionName(count),false,Vector(Variable(statements)))),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(setters), IdName(statementsCount)),Selections(Set()),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(setters -> Variable(setters), statementsCount -> Variable(statementsCount),   FRESHID269 -> Equals(Variable(statementsCount),Parameter(  AUTOINT3,Integer))),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(  UNNAMED478),(IdName(setters),IdName(  UNNAMED495)),OUTGOING,List(RelTypeName(vparameters)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED289),(IdName(setters),IdName(  UNNAMED300)),OUTGOING,List(RelTypeName(vBody)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED337),(IdName(  UNNAMED300),IdName(  UNNAMED354)),OUTGOING,List(RelTypeName(vstatements)),SimplePatternLength), PatternRelationship(IdName(  UNNAMED405),(IdName(  UNNAMED354),IdName(  UNNAMED408)),OUTGOING,List(),SimplePatternLength)),Set(IdName(  UNNAMED495), IdName(  UNNAMED354), IdName(  UNNAMED300), IdName(  UNNAMED408), IdName(setters), IdName(  rankf@668)),Set(IdName(setters), IdName(statementsCount), IdName(  FRESHID269)),Selections(Set(Predicate(Set(IdName(  FRESHID269)),Variable(  FRESHID269)), Predicate(Set(IdName(  UNNAMED289), IdName(  UNNAMED405)),Not(Equals(Variable(  UNNAMED289),Variable(  UNNAMED405)))), Predicate(Set(IdName(  UNNAMED354)),In(Property(Variable(  UNNAMED354),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING5,String))))), Predicate(Set(IdName(  rankf@668)),In(Property(Variable(  rankf@668),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING8,String))))), Predicate(Set(IdName(  UNNAMED300)),HasLabels(Variable(  UNNAMED300),List(LabelName(Block)))), Predicate(Set(IdName(  rankf@668)),HasLabels(Variable(  rankf@668),List(LabelName(MethodDeclaration)))), Predicate(Set(IdName(  UNNAMED408)),HasLabels(Variable(  UNNAMED408),List(LabelName(Assignment)))), Predicate(Set(IdName(  UNNAMED354)),HasLabels(Variable(  UNNAMED354),List(LabelName(ExpressionStatement)))), Predicate(Set(IdName(  UNNAMED408)),In(Property(Variable(  UNNAMED408),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING6,String))))), Predicate(Set(IdName(  UNNAMED300)),In(Property(Variable(  UNNAMED300),PropertyKeyName(projectID)),ListLiteral(List(Parameter(  AUTOSTRING4,String))))), Predicate(Set(IdName(  UNNAMED337), IdName(  UNNAMED405)),Not(Equals(Variable(  UNNAMED337),Variable(  UNNAMED405)))))),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(  res@517 -> Variable(  res@517),   rankf@668 -> Variable(  rankf@668),   ruleCoef@789 -> Parameter(  AUTOINT9,Integer)),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(  res@517), IdName(  rankf@668), IdName(  ruleCoef@789)),Selections(Set()),Vector(),Set(),Set(),List()),AggregatingQueryProjection(Map(res.nomFichier -> Property(Variable(  res@517),PropertyKeyName(nomFichier)), res.lineNum -> Property(Variable(  res@517),PropertyKeyName(lineNum)), 1 -> Parameter(  AUTOINT10,Integer)),Map(),QueryShuffle(List(),None,None)),None)))))))

```

**Comment by sherfert** [comment]
Thanks for reporting this, we're on it.

**Comment by OliviaYtterbrink** [comment]
This has been fixed by #10121 which is in the 3.2.6 release.
